src/test/lib/scheduler/scheduler_test.cpp

```diff
@@ -66,23 +68,25 @@ class SchedulerTest : public BaseTest {
     task3->schedule();
     task1->schedule();
     task2->schedule();
+
+    Hyrise::get().scheduler()->finish();
```

Comment:
The `BaseTest` dtor calls `hyrise::reset()`, which itself calls the scheduler to finish. Do we need the call in the test scope or what do we expect here?

Reply:
We check the counter in some tests. So we need to finish before reading the counter. We could also use a vector of tasks and use `schedule_and_wait` or just `wait` but it should be the same in the end.

Now that you say it, using a vector and using `wait_for_tasks` sounds better than this implicit finish to make sure they finished.

Reply:
Changed finish to `Hyrise::get().scheduler()->wait_for_tasks({task1, task2, task3});`

What do you think?